test/lit/passes/j2cl-merge-itables.wast

```diff
@@ -0,0 +1,222 @@
+
+    ;; CHECK:       (type $2 (func))
+
+    ;; CHECK:       (type $function (func))
+
+    ;; CHECK:       (type $Object.vtable (sub (struct (field structref))))
+    (type $Object.vtable (sub (struct)))
+    ;; CHECK:       (type $SubObject.vtable (sub $Object.vtable (struct (field structref) (field (ref $function)))))
+
+    ;; CHECK:       (type $Object.itable (struct (field structref)))
+    (type $Object.itable (struct
+		  (field (ref null struct))))
+
+    (type $Object (sub (struct
+      (field $vtable (ref $Object.vtable))
+      (field $itable (ref $Object.itable)))))
+
+    ;; SubObject class type definitions.
+    (type $SubObject.vtable (sub $Object.vtable (struct
+		  (field (ref $function)))))
+
+    (type $SubObject (sub $Object (struct
+      (field $vtable (ref $SubObject.vtable))
+      (field $itable (ref $Object.itable)))))
+
+    (type $function (func))
+  )
+
+  ;; Object vtable and itable initialization.
+  ;; CHECK:      (global $Object.itable (ref $Object.itable) (struct.new_default $Object.itable))
+
+  ;; CHECK:      (global $SubObject.vtable (ref $SubObject.vtable) (struct.new $SubObject.vtable
+  ;; CHECK-NEXT:  (ref.null none)
+  ;; CHECK-NEXT:  (ref.func $SubObject.f)
+  ;; CHECK-NEXT: ))
+
+  ;; CHECK:      (global $SubObject.itable (ref $Object.itable) (global.get $Object.itable))
+
+  ;; CHECK:      (global $Object.vtable (ref $Object.vtable) (struct.new $Object.vtable
+  ;; CHECK-NEXT:  (ref.null none)
+  ;; CHECK-NEXT: ))
+  (global $Object.vtable (ref $Object.vtable)
+    (struct.new $Object.vtable))
+  (global $Object.itable (ref $Object.itable)
+    (struct.new_default $Object.itable))
+
+  ;; SubObject vtable and itable initialization. Shared empty itable.
+  (global $SubObject.vtable (ref $SubObject.vtable)
+    (struct.new $SubObject.vtable (ref.func $SubObject.f)))
+  (global $SubObject.itable (ref $Object.itable)
+    (global.get $Object.itable))
+
+  ;; CHECK:      (func $SubObject.f (type $function)
+  ;; CHECK-NEXT:  (nop)
+  ;; CHECK-NEXT: )
+  (func $SubObject.f
+    (type $function)
+  )
+
+  ;; CHECK:      (func $usages (type $2)
+  ;; CHECK-NEXT:  (local $o (ref null $SubObject))
+  ;; CHECK-NEXT:  (local.set $o
+  ;; CHECK-NEXT:   (struct.new $SubObject
+  ;; CHECK-NEXT:    (global.get $SubObject.vtable)
+  ;; CHECK-NEXT:    (global.get $SubObject.itable)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.get $SubObject.vtable 1
+  ;; CHECK-NEXT:    (struct.get $SubObject $vtable
+  ;; CHECK-NEXT:     (local.get $o)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.get $SubObject.vtable 0
+  ;; CHECK-NEXT:    (struct.get $SubObject $vtable
+  ;; CHECK-NEXT:     (local.get $o)
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $usages
+    (local $o (ref null $SubObject))
+    (local.set $o
+			(struct.new $SubObject
```

Comment:
Probably it is in the infra but: are you using this format with a lot of empty space or does the infra require them?

Reply:
I had the correct setting for tabs to 2 spaces but I guess I had the wrong configuration for .wast files and was inserting tabs instead of spaces.